### PR TITLE
Added solution with simple DTO

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

**Review by MichaelShevchenko — APPROVED** [review]
Well done
Только пустая строка в конце web.xml. ну и не совсем уверен для чего он нам сейчас нужен

File: web/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
```
Comment (RomanDubovskyi) [inline web/WEB-INF/web.xml]: do we need web.xml?

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -43,6 +43,16 @@ public User add(User user) {
         return user;
     }
 
+    @Override
+    public User getById(Long id) {
+        log.info("Calling method user by id = " + id);
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM User u WHERE u.id = :id", User.class)
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java]: let's use session get instead

File: src/main/java/spring/intro/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @GetMapping("/user/{id}")
+    public UserResponseDto get(@PathVariable Long id) {
+        User user = userService.getById(id);
+        return mapDto(user);
+    }
+
+    @GetMapping("/user/all")
+    public List<UserResponseDto> getAll() {
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/controllers/UserController.java:41]: move user url mapping to class lever

File: src/main/java/spring/intro/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        return new RedirectView("/user/all");
+    }
+
+    @GetMapping("/user/{id}")
+    public UserResponseDto get(@PathVariable Long id) {
+        User user = userService.getById(id);
+        return mapDto(user);
+    }
+
+    @GetMapping("/user/all")
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/controllers/UserController.java]: let's create our url address according to best practices from checklist 
https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_32.html

File: web/WEB-INF/web.xml (hunk 1 above)
Comment (Nokafan) [reply web/WEB-INF/web.xml]: No, we don't need it, removed already.

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/spring/intro/dao/impl/UserDaoImpl.java]: Changed to get()

File: src/main/java/spring/intro/controllers/UserController.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/spring/intro/controllers/UserController.java:41]: Changed to @RequestMapping() for class

File: src/main/java/spring/intro/controllers/UserController.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/spring/intro/controllers/UserController.java]: Fixed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**